rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/EmbeddedProject.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Siemens AG, 2019. Part of the SW360 Portal Project.
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.sw360.rest.resourceserver.project;
+
+import org.eclipse.sw360.datahandler.thrift.projects.Project;
+import org.springframework.hateoas.core.Relation;
+
+/**
+ * 
+ * This class is used for json customization for embedded project.
+ * 
+ * @author smruti.sahoo@siemens.com
+ *
+ */
+@Relation(collectionRelation = "sw360:projects")
+public class EmbeddedProject extends Project {
```

Comment:
Why does a project need this `EmbeddedProject` construction, but no other resource has something similar (although embedded users or licenses are implemented?)

Reply:
With the current implementation, its little difficult to blacklist the boolean fields (like `enableSvm,` `enableVulnerabilitiesDisplay` etc.) only in the project listing through the `api/projects` endpoint and white list it, when requesting for a single project through the `api/projects/:id` endpoint . That's why I have to add a subclass for the project to achieve the same. We can add the same implementation for other resources like users or licences, may be later, when we face same type of situation. Please let me know , if you have any other idea.